src/App.ts

```diff
@@ -0,0 +1,29 @@
+import express from 'express';
+import type { Routers } from './routers';
+
+/**
+ * The main server application.
+ */
+export class App {
```

Comment:
=

> We can think of `app.js` as the primary logic for setting up this application -- loading ports, etc -- and `index.js` as the entrypoint that determines when and how to start and stop the application.

I think that's what the App class here does. I guess whether to use a class or anonymous class or singleton or whatever pattern is less important. I completely agree the wrapper for Router is of no advantage.

Reply:
> A benefit of exposing an express app directly is that, with a wrapper, we need to basically re-implement every express method we want to expose (right now, and probably forever, it is just `listen` but who knows what we might want in future!)

We control both the index.js and the app.js or App.js, so we can decide how much and what methods need exposure at what level. I generally default to less visibility, less mutable state, etc., and then expand out from there rather than starting wide open and then attempting to narrow later. But by all means, when in Rome, do as the Romans do :smile:

Reply:
Understood!  I do think one of the benefits of using a framework like express is they have already thought about those API decisions (so we don't need to write a wrapper for the express `listen` -- we can just expose the express `listen`).